### ISSUE TITLE
[no-dupe-class-members] Rule doesn't handle class properties

### ISSUE THREAD

**Issue by odaper** [typescript-eslint/typescript-eslint#618]
I have configured the following .eslintrc to validate my typescript files:

```
module.exports = {
  env: {
    node: true,
    es6: true,
  },
  extends: ['plugin:@typescript-eslint/recommended'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    sourceType: 'module',
    typescript: true,
    modules: true,
    ecmaVersion: 6,
    ecmaFeatures: {
      experimentalObjectRestSpread: true,
    },
  },
  plugins: ['@typescript-eslint'],
  rules: {
    'no-duplicate-imports': 'error',
    'no-var': 'error',
    'no-dupe-keys': 'error',
    'no-duplicate-case': 'error',
    'no-dupe-class-members': 'error',
    indent: ['error', 4],
    'linebreak-style': ['error', 'unix'],
    quotes: ['error', 'single'],
    semi: ['error', 'always'],
  },
};
```
When my typescript file contains duplicated property name like:
```
 export class TestClassViewComponent  {

  selectedItems: TestClassModel[];
  selectedItems: TestClassModel[];
....
```
No error will be thrown when running the linting process.
Is "no-dupe-class-members" supported in typescript files?
Many thanks

**Comment by bradzacher** [comment]
this is expected. eslint core doesn't support class properties as they're not yet part of the JS spec.

We recommend just turning off this rule for typescript files because typescript does its own checks for duplicate class members.

```ts
  export class TestClassViewComponent  {
    selectedItems: TestClassModel[];
    selectedItems: TestClassModel[];
//  ^^^^^^^^^^^^^ Duplicate identifier 'selectedItems'
  }
```

Similar to #291 - I'll leave it open in case anyone really wants to implement a fix for this, but it's bottom of our priority because of our recommendation not to use the rule.

**Comment by armano2** [comment]
this issue has been fixed upstream, eslint is now supporting class properties, see [Playground](https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=KYDwDg9gTgLgBAYwDYEMDOa4BVhpgYVQwDUBLYAd3wgFtIA7Ye+AbwFgAoOONYJYBDGAATAJJCaaAFzZcBImgCyEYXwDaAXQDcnbr36CR44JJk48hdEpXrtnAL5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaAE1gAdFCBjeAQ2WUIFtEmAjRaVDST6faJHBgAviBFA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)
